projects/sbb-esta/angular-business/dialog/src/dialog/dialog-container/dialog-container.component.ts

```diff
@@ -0,0 +1,231 @@
+import { AnimationEvent } from '@angular/animations';
+import { FocusTrap, FocusTrapFactory } from '@angular/cdk/a11y';
+import {
+  BasePortalOutlet,
+  CdkPortalOutlet,
+  ComponentPortal,
+  TemplatePortal
+} from '@angular/cdk/portal';
+import { DOCUMENT } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ComponentRef,
+  ElementRef,
+  EmbeddedViewRef,
+  EventEmitter,
+  HostBinding,
+  HostListener,
+  Inject,
+  Optional,
+  ViewChild,
+  ViewEncapsulation
+} from '@angular/core';
+
+import { DIALOG_ANIMATIONS } from '../dialog-animations';
+import { DialogConfig } from '../dialog-config';
+
+/**
+ * Throws an exception for the case when a ComponentPortal is
+ * attached to a DomPortalOutlet without an origin.
+ * @docs-private
+ */
+export function throwDialogContentAlreadyAttachedError() {
+  throw Error('Attempting to attach dialog content after content is already attached');
+}
+
+/**
+ * Internal component that wraps user-provided dialog content.
+ * @docs-private
+ */
+@Component({
+  selector: 'sbb-dialog-container',
+  templateUrl: 'dialog-container.component.html',
+  styleUrls: ['dialog-container.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  animations: [DIALOG_ANIMATIONS.slideDialog]
+})
+export class DialogContainerComponent extends BasePortalOutlet {
+  /** The portal outlet inside of this container into which the dialog content will be loaded. */
+  @ViewChild(CdkPortalOutlet, { static: true }) portalOutlet: CdkPortalOutlet;
+
+  @HostBinding('class.sbb-dialog-container')
+  containerClass = true;
+
+  @HostBinding('attr.tabindex')
+  tabIndex = '-1';
+
+  @HostBinding('attr.aria-modal')
+  arialModal = 'true';
+
+  @HostBinding('attr.id')
+  get dialogContainerID() {
+    return this.id;
+  }
+
+  @HostBinding('attr.role')
+  get role() {
+    return this.config.role;
+  }
+
+  @HostBinding('attr.aria-labelledby')
+  get ariaLabelledbyAttr() {
+    return this.config.ariaLabel ? null : this.ariaLabelledBy;
```

Comment:
I believe this is a mistake in the lightbox. This should be `return this.config.ariaLabelledBy ? null : this.ariaLabelledBy;`.

Reply:
`this.config.ariaLabelledBy` does not exist at the moment. Should I add it?

Reply:
Sorry, should have checkd first. I'm mistaken. Forget about this comment. 🙇